electrode_models/nucleation_single_particle.py

```diff
@@ -0,0 +1,291 @@
+        if self.name=='anode':
+            self.i_ext_flag = -1
+        elif self.name=='cathode':
+            self.i_ext_flag = 1
+        else:
+            raise ValueError("Electrode must be an anode or a cathode.")
+
+        # Store the species index of the Li ion in the Cantera object for the 
+        # electrolyte phase:
+        self.index_Li = self.elyte_obj.species_index(inputs['mobile-ion'])
+
+        # Electrode thickness and inverse thickness:
+        self.dy = inputs['thickness']
+        self.dyInv = 1/self.dy
+
+        self.eps_solid = inputs['eps_solid']
+        self.eps_elyte = 1 - self.eps_solid
+        self.eps_oxide = 0
+        
+        self.V_part = 4./3. * np.pi * (inputs['r_p']/ 2)**3  # particle volume [m3]
+        self.A_part = 4. * np.pi * (inputs['r_p'] / 2)**2    # particle surface area [m2]
+        self.A_init = self.eps_solid * self.A_part / self.V_part 
+        self.th_oxide= inputs['th_oxide']
+        self.A_oxide = np.pi* inputs['d_part']**2/4.
+        self.V_oxide = 2./3. * np.pi* (inputs['d_part']/2.)**2 * self.th_oxide
+
+        # For some models, the elyte thickness is different from that of the 
+        # electrode, so we specify is separately:
+        self.dy_elyte = self.dy
+
+        # Phase volume fractions
+        self.eps_solid = inputs['eps_solid']
+        self.eps_elyte = 1 - self.eps_solid
+        self.eps_oxide = 0
+
+        # Electrode-electrolyte interface area, per unit geometric area.
+        # This calculation assumes spherical particles of a single radius, with 
+        # no overlap.
+        self.A_surf_ratio = (3*self.eps_solid*self.dy/inputs['r_p'])
+
+        # Inverse double layer capacitance, per unit interfacial area.
+        self.C_dl_Inv = 1/inputs['C_dl']
+
+        # Microstructure-based transport scaling factor, based on Bruggeman 
+        # coefficient of -0.5:
+        self.elyte_microstructure = self.eps_elyte**1.5
+        
+        # SV_offset specifies the index of the first SV variable for the 
+        # electode (zero for anode, n_vars_anode + n_vars_sep for the cathode)
+        self.SV_offset = offset
+
+        # Determine the electrode capacity (Ah/m2)
+
+        # Max concentration of stored ion (intercalated Li)
+        # Save initial X
+        #X_o = self.bulk_obj.X 
+        # Set object concentration to fully lithiated:
+        #self.bulk_obj.X = inputs['stored-ion']['name']+':1.0' 
+        # Concentration of stored Li, per unit volume of intercalation phase:
+        #Conc = self.bulk_obj[inputs['stored-ion']['name']].concentrations[0]
+        
+        self.capacity = (inputs['stored-species']['charge']*ct.faraday
+                 *self.eps_elyte)*inputs['thickness']/(3600
+                 *inputs['stored-species']['MW']) 
+
+        # Return Cantera object composition to original value:
+        #self.bulk_obj.X = X_o
+        
+        # Number of state variables: electrode potential, electrolyte, porosity 
```

Comment:
The `n_vars` code is correct, but this comment is not.

1. Replace "porosity" with "LiO2 volume fraction."
2. Remove "electrode composition."